src/term/mod.rs

```diff
@@ -274,6 +281,11 @@ impl Book {
   pub fn is_def_name_generated(&self, def_id: DefId) -> bool {
     self.def_names.name(&def_id).map_or(false, |Name(name)| name.contains('$'))
   }
+
+  /// Checks if the definition of the given DefId is a built-in
+  pub fn is_builtin(&self, def_id: DefId) -> bool {
+    matches!(self.defs.get(&def_id).unwrap().rules[0].origin, Origin::Builtin)
```

Comment:
I'd suggest deriving PartialEq for Origin and using a `...origin == Origin::BuiltIn`. I think it's a bit more intuitive to read